src/app/core/entity/entity-remove.service.spec.ts

```diff
@@ -263,38 +276,292 @@ describe("EntityRemoveService", () => {
+    await testAction(
+      "delete",
+      entity,
+      [entity, ref1],
+      [RelatedEntity.create("ref", { refAggregate: [] })],
+    );
+  });
+  fit("should not cascade anonymize the 'aggregate'-type entity that only references the entity user acts on but ask user", async () => {
+    const entity = RelatedEntity.create("entity user acts on");
+    const ref1 = RelatedEntity.create("ref", {
+      refAggregate: [entity.getId()],
+    });
+
+    // TODO: require manual review / confirmation by the user?
+    await testAction(
+      "anonymize",
+      entity,
+      [entity, ref1],
+      [RelatedEntity.create(undefined), ref1],
+    );
+  });
+
+  fit("should not cascade delete the 'aggregate'-type entity that still references additional other entities but remove id", async () => {
+    const entity = RelatedEntity.create("entity user acts on");
+    const entity2 = RelatedEntity.create("other entity");
+
+    const ref1 = RelatedEntity.create("ref", {
+      refAggregate: [entity.getId(), entity2.getId()],
+    });
+
+    await testAction(
+      "delete",
+      entity,
+      [entity, entity2, ref1],
+      [
+        entity2,
+        RelatedEntity.create("ref", { refAggregate: [entity2.getId()] }),
       ],
     );
   });
+  fit("should not cascade anonymize the 'aggregate'-type entity that still references additional other entities but ask user", async () => {
+    const entity = RelatedEntity.create("entity user acts on");
+    const entity2 = RelatedEntity.create("other entity");
+
+    const ref1 = RelatedEntity.create("ref", {
+      refAggregate: [entity.getId(), entity2.getId()],
+    });
+
+    await testAction(
+      "anonymize",
+      entity,
+      [entity, entity2, ref1],
+      [
+        RelatedEntity.create(undefined),
+        entity2,
+        // TODO: require manual review / confirmation by the user?
+        ref1,
+      ],
+    );
+  });
+
+  fit("should undo delete for all cascadingly affected entities", fakeAsync(() => {
+    const onSnackbarAction = new Subject<void>();
+    mockSnackBarRef.onAction.and.returnValue(onSnackbarAction.asObservable());
+
+    const entity = RelatedEntity.create("entity user acts on");
+    const ref1 = RelatedEntity.create("ref-1", {
+      refComposite: [entity.getId()],
+    });
+    const ref2 = RelatedEntity.create("ref-2", {
+      refAggregate: [entity.getId()],
+    });
+
+    testAction(
+      "delete",
+      entity.copy(),
+      [entity.copy(), ref1.copy(), ref2.copy()],
+      [RelatedEntity.create("ref-2", { refAggregate: [] })],
+    );
+    tick();
+
+    onSnackbarAction.next();
+    onSnackbarAction.complete();
+    tick();
+
+    const actualEntitiesAfterUndo = // @ts-ignore
+      (service.entityMapper as MockEntityMapperService).getAllData();
+    expectEntitiesToMatch(actualEntitiesAfterUndo, [entity, ref1, ref2], true);
+  }));
+
+  // TODO: test Note with attendance (?)
+  // TODO: test for entity with two properties referencing the same entity type
```

Comment:
the way the search for affected entities is implemented, it might possibly be detected twice and then be edited twice - if badly implemented, this could lead to things being overwritten or conflicts to occur.